external/optiga_config.h

```diff
@@ -0,0 +1,125 @@
+    /** @brief OPTIGA CRYPT RSA SSA with SHA512 as digest feature enable/disable macro */      
+    #define OPTIGA_CRYPT_RSA_SSA_SHA512_ENABLED      
+    /** @brief OPTIGA CRYPT symmetric encrypt feature enable/disable macro */
+    #define OPTIGA_CRYPT_SYM_ENCRYPT_ENABLED
+    /** @brief OPTIGA CRYPT symmetric decrypt feature enable/disable macro */
+    #define OPTIGA_CRYPT_SYM_DECRYPT_ENABLED
+    /** @brief OPTIGA CRYPT HMAC feature enable/disable macro */
+    #define OPTIGA_CRYPT_HMAC_ENABLED
+    /** @brief OPTIGA CRYPT HKDF feature enable/disable macro */
+    #define OPTIGA_CRYPT_HKDF_ENABLED
+    /** @brief OPTIGA CRYPT symmetric generate key feature enable/disable macro */
+    #define OPTIGA_CRYPT_SYM_GENERATE_KEY_ENABLED
+    /** @brief OPTIGA CRYPT generate auth code feature enable/disable macro */
+    #define OPTIGA_CRYPT_GENERATE_AUTH_CODE_ENABLED
+    /** @brief OPTIGA CRYPT HMAC verify feature enable/disable macro */
+    #define OPTIGA_CRYPT_HMAC_VERIFY_ENABLED
+    /** @brief OPTIGA CRYPT clear AUTO state feature enable/disable macro */
+    #define OPTIGA_CRYPT_CLEAR_AUTO_STATE_ENABLED   
+
+    /** @brief OPTIGA COMMS shielded connection feature.
+     *         To disable the feature, undefine the macro
+     */
+    #define OPTIGA_COMMS_SHIELDED_CONNECTION
+
+    /** @brief Default reset protection level for OPTIGA CRYPT and UTIL APIs */
+    #define OPTIGA_COMMS_DEFAULT_PROTECTION_LEVEL           OPTIGA_COMMS_FULL_PROTECTION
+    //#define OPTIGA_COMMS_DEFAULT_PROTECTION_LEVEL           OPTIGA_COMMS_NO_PROTECTION
+
+  /** @brief Default reset type in optiga_comms_open.             \n
+     *         Cold Reset - (0) : This is applicable if the host platform has GPIO option for RST and VDD.    \n
+     *         Soft Reset - (1) : This is applicable if the host platform doesn't have GPIO options for VDD and RST.  \n
+     *         Warm Reset - (2) : This is applicable if the host platform doesn't have GPIO option for VDD. \n
+     *         Any other value will lead to error
+     */
+    #define OPTIGA_COMMS_DEFAULT_RESET_TYPE     (1U)
+    
+    /** @brief NULL parameter check.
+     *         To disable the check, undefine the macro
+     */
+    #define OPTIGA_LIB_DEBUG_NULL_CHECK
+    /** @brief Maximum number of instance registration */
+    #define OPTIGA_CMD_MAX_REGISTRATIONS                (0x06)
+    /** @brief Maximum buffer size required to communicate with OPTIGA */
+    #define OPTIGA_MAX_COMMS_BUFFER_SIZE                (0x615) //1557 in decimal
+
+    /** @brief Macro to enable logger \n
+    * Enable macro OPTIGA_LIB_ENABLE_UTIL_LOGGING for Util Service layer logging     \n
+    * Enable macro OPTIGA_LIB_ENABLE_CRYPT_LOGGING for Crypt Service layer logging     \n
+    * Enable macro OPTIGA_LIB_ENABLE_CMD_LOGGING for Command layer logging     \n
+    * Enable macro OPTIGA_LIB_ENABLE_COMMS_LOGGING for Communication layer logging     */
+    #define OPTIGA_LIB_ENABLE_LOGGING
```

Comment:
Should logging be enabled, or should you enable that when you develop? Or perhaps make it conditional on `NDEBUG`?

Reply:
Removing this saves 4440 bytes! Good catch. I haven't seen any logs and have not been missing them, ok to disable for me.